eventDataScript.js

```diff
@@ -0,0 +1,296 @@
+    }
+    $("#eventSpecificTeamTable").html('<tr><th><b>Team Number</b></th> <th><b>Team Name</b></th> </tr>');
+
+    var teamsJSON = JSON.parse(getCookie(event_data_cookie_name(season,currEvent)));
+    if (teamsJSON == '{}') return;
+    const pairs = Object.entries(teamsJSON);
+    for (var i = 0; i < pairs.length; i++)
+    {
+        var teamNumber = pairs[i][0];
+        var teamName = pairs[i][1];
+
+        var newTRHtml = "<tr><td>" + teamNumber + 
+        "</td><td>" + teamName + "</td></tr>";
+
+        if ($('#eventSpecificTeamTable').html().indexOf(newTRHtml) < 0 &&
+        (teamNumber != '' || teamName != ''))
+        {
+            $('#eventSpecificTeamTable').append(newTRHtml);
+        }
+    }
+}
+
+//onclick for deleting a specific team based on number
+function deleteSpecificTeam()
+{
+    var input = $('#deleteSpecificTeamTextBox').val().trim(); //should be team number
+    if (input === '') return;
+
+    var season = getCookie('currCheckedSeason');
+    var currEvent = getCurrEvent(); 
+    var cname = event_data_cookie_name(season,currEvent);
+    var teamsJSON = JSON.parse(getCookie(cname));
+
+    delete teamsJSON[input];
+    setCookie(cname,JSON.stringify(teamsJSON),750);
+    setTeamTable(season,currEvent);
+    $("#deleteSpecificTeamTextBox").val("").focus();
+}
+
+$(document).ready(function(){
+    loadPage();
+
+    const Enter_key_code = 13;
+    $('#newEventInputBox').keypress(function(e){
+        if (e.keyCode == Enter_key_code) addNewEvent();
+    });
+
+    //event specific teams code
+    $('#teamName').keypress(function(e){
+        if (e.keyCode == Enter_key_code) $('#teamSaveButton').click();
+    });
+
+    $('#deleteSpecificTeamTextBox').keypress(function(e){
+        if (e.keyCode == Enter_key_code) deleteSpecificTeam();
+    });
+
+    $('#teamSaveButton').click(function(){
+        var currSeason = getCookie('currCheckedSeason');
+        var currEvent = getCurrEvent();
+        var teamNumber = parseInt($("#teamNumber").val().trim());
+        var teamName = $("#teamName").val().trim();
+        var teams = {};
+
+        if (teamNumber === "" || teamName === "")
+        {
+            $('#teamConfirmationBox').html("<i>Please fill in both fields.<\i>");
+            return;
+        }
+
+        if (currSeason === ' ')
+        {
+            $("#teamConfirmationBox").html("<i>Please check a season.<\i>");
+            return;
+        }
+
+        if (currEvent === ' ')
+        {
+            $("#teamConfirmationBox").html("<i>Please check an event.<\i>");
+            return;
+        }
+
+        var cname = event_data_cookie_name(currSeason,currEvent);
+        var teams;
+        var raw_stored_data = getCookie(cname).trim();
+        if (!raw_stored_data == '') //if cookie is stored yet
+        {
+            teams = JSON.parse(raw_stored_data);
+        } else //cookie not stored yet
+        {
+            teams = {};
+        }
```

Comment:
just set the default up at the start:

```js
var raw_stored_data = getCookie(cname).tim();
var teams = {};
if (raw_stored_data.length > 0) //if cookie is stored yet
{
    teams= JSON.parse(raw_stored_data);
}
```

Also, `(!raw_stored_data == '')` probably isn't doing what you think.
`!raw_stored_data` will convert `raw_stored_data` to a bool then not it (a string other than empty = true, and empty string = false), ...
that's functionally equivalent to, `(raw_stored_data == "")?false:true`
then, it will convert the `''` at the end to a bool (which will be false) for the comparison, so what you've coded is:
`if( (raw_stored_data == "")?false:true == false)`, this is doing what you wanted, but it would read a lot better with either:
`if( raw_stored_data != '')` or `if(raw_stored_data.length > 0)`